frontend/cypress/integration/basic-spec.ts

```diff
@@ -0,0 +1,84 @@
+import { BigNumber } from 'ethers';
+import { expect } from 'chai';
+
+describe('Collateral auctions', function () {
+    // Set the private key to one of the deterministic hardhat keys.
+    // see https://hardhat.org/hardhat-network#running-stand-alone-in-order-to-support-wallets-and-other-software
+    const KEEPER_WALLET_PRIVATE_KEY = '0x701b615bbdfb9de65240bc28bd21bbc0d996645a3dd57e7b12bc2bdf6f192c82';
+    const URL_TO_VISIT = `${process.env.FRONTEND_URL || 'localhost:3000'}/collateral?network=localhost`;
+    let daiBalanceBefore: Record<string, BigNumber>;
+    let daiBalanceAfter: Record<string, BigNumber>;
+
+    function testSwapProfit() {
+        cy.visit(URL_TO_VISIT);
+        cy.window()
+            .its('$nuxt')
+            .then(function (nuxt) {
+                nuxt.$store.dispatch('wallet/createWalletFromPrivateKey', KEEPER_WALLET_PRIVATE_KEY);
+            });
+        cy.get('table').first({ timeout: 140000 }).should('contain.text', 'ETH');
+
+        cy.window()
+            .its('$nuxt')
+            .then(function (nuxt) {
+                daiBalanceBefore = nuxt.$store.getters['wallet/walletBalances'];
+            });
+
+        cy.get('a').eq(1).should('contain.text', 'Participate').click();
+        cy.get('button.Button').eq(3).should('contain.text', 'Directly swap').click();
```

Comment:
So adding a new unrelated button to the page can break the test?  That's called fragile tests. We don't want that. Good test should be as readable as specification of what should happen without any magic in-depth knowledge of selectors involved.

Based on this: please use more specific selectors or add new classes to the buttons in question, eg `SwapButton`

here and everywhere else

Reply:
Another approach will be to use contains directly, without specifying a selector, [according to the docs](https://docs.cypress.io/api/commands/contains#Syntax):

```suggestion
        cy.contains('Directly swap').click();
```

But it doesn't really work for all buttons, since some buttons contains content-specific values, so adding readable class names is more generic approach

Reply:
Turns out that `contains` accepts regexps which makes selection easier 

( initial problem that caused the hack was that it selects the first element that contains the argument )